src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -172,14 +172,20 @@ def add_electrode_groups(recording: SpikeInterfaceRecording, nwbfile: pynwb.NWBF
         metadata = dict()
     if "Ecephys" not in metadata:
         metadata["Ecephys"] = dict()
+
+    if "group_name" in checked_recording.get_property_keys():
+        group_names = np.unique(checked_recording.get_property("group_name"))
+    else:
+        group_names = np.unique(checked_recording.get_channel_groups()).astype("str", copy=False)
```

Comment:
why call unique?

Reply:
The function returns the name of the group for each of the channels. But we only need to create a group object for each of the unique names.

Reply:
ah, ok, I see